packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -98,219 +103,207 @@ export interface TooltipProps
  * @accessibility
  * Implements [ARIA Tooltip](https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip) design pattern.
  */
-export default class Tooltip extends AutoControlledComponent<TooltipProps, TooltipState> {
-  static displayName = 'Tooltip'
-
-  static className = 'ui-tooltip'
-
-  static slotClassNames: TooltipSlotClassNames = {
-    content: `${Tooltip.className}__content`,
-  }
-
-  static Content = TooltipContent
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      as: false,
-      content: false,
+const Tooltip: React.FC<TooltipProps> &
+  FluentComponentStaticProps<TooltipProps> & {
+    Content: typeof TooltipContent
+    slotClassNames: TooltipSlotClassNames
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext)
+  const { setStart, setEnd } = useTelemetry(Tooltip.displayName, context.telemetry)
+  setStart()
+
+  const {
+    accessibility,
+    align,
+    children,
+    content,
+    mountNode,
+    mouseLeaveDelay,
+    offset,
+    pointing,
+    position,
+    target,
+    trigger,
+  } = props
+
+  const [open, setOpen] = useAutoControlled({
+    defaultValue: props.defaultOpen,
+    value: props.open,
+
+    initialValue: false,
+  })
+
+  const contentRef = React.useRef<HTMLElement>()
+  const pointerTargetRef = React.useRef<HTMLDivElement>()
+  const triggerRef = React.useRef<HTMLElement>()
+
+  const closeTimeoutId = React.useRef<number>()
+  // TODO: Consider `getOrGenerateIdFromShorthand()` as hook and make it SSR safe
```

Comment:
Ups

Reply:
I want to do it later as we use this function in more components (`Dialog` at least)